plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.tsx

```diff
@@ -139,90 +78,47 @@ export const OwnershipCard = ({
   variant?: InfoCardVariants;
   entityFilterKind?: string[];
 }) => {
+  const classes = useStyles();
   const { entity } = useEntity();
-  const catalogApi = useApi(catalogApiRef);
-  const catalogLink = useRouteRef(catalogIndexRouteRef);
-
-  const {
-    loading,
-    error,
-    value: componentsWithCounters,
-  } = useAsync(async () => {
-    const kinds = entityFilterKind ?? ['Component', 'API'];
-    const entitiesList = await catalogApi.getEntities({
-      filter: {
-        kind: kinds,
-      },
-      fields: [
-        'kind',
-        'metadata.name',
-        'metadata.namespace',
-        'spec.type',
-        'relations',
-      ],
-    });
-
-    const ownedEntitiesList = entitiesList.items.filter(component =>
-      isOwnerOf(entity, component),
-    );
-
-    const counts = ownedEntitiesList.reduce(
-      (acc: EntityTypeProps[], ownedEntity) => {
-        const match = acc.find(
-          x =>
-            x.kind === ownedEntity.kind &&
-            x.type === (ownedEntity.spec?.type ?? ownedEntity.kind),
-        );
-        if (match) {
-          match.count += 1;
-        } else {
-          acc.push({
-            kind: ownedEntity.kind,
-            type: ownedEntity.spec?.type?.toString() ?? ownedEntity.kind,
-            count: 1,
-          });
-        }
-        return acc;
-      },
-      [],
-    );
-
-    // Return top N (six) entities to be displayed in ownership boxes
-    const topN = counts.sort((a, b) => b.count - a.count).slice(0, 6);
-
-    return topN.map(topOwnedEntity => ({
-      counter: topOwnedEntity.count,
-      type: topOwnedEntity.type,
-      name: topOwnedEntity.type.toLocaleUpperCase('en-US'),
-      queryParams: getQueryParams(entity, topOwnedEntity),
-    })) as Array<{
-      counter: number;
-      type: string;
-      name: string;
-      queryParams: string;
-    }>;
-  }, [catalogApi, entity]);
-
-  if (loading) {
-    return <Progress />;
-  } else if (error) {
-    return <ResponseErrorPanel error={error} />;
-  }
+  const isGroup = entity.kind === 'Group';
+  const [relationsType, setRelationsType] = useState('direct');
+  const renderedGrid =
+    relationsType !== 'direct' && isGroup
+      ? aggregatedRelationsGrid(entity, entityFilterKind)
+      : directRelationsGrid(entity, entityFilterKind);
```

Comment:
Not really worth making functions for these.

In the code below you could just do

```tsx
{(relationsType !== 'direct' && isGroup)
  ? <DirectRelationsGrid entity={entity} entityFilterKind={entityFilterKind} />
  : <...>
}
```

however, as noted elsewhere, I think the better solution is to just make them into one single component that has the relation type prop because the code paths are basically 100% the same

Reply:
updated @freben